### PR TITLE
refactor is_equal not to compare len at each loop

### PR DESCRIPTION
**Pull request by gaetbout**
is_equal optimization
remove some mut at the interface level to do it within the fn

### CODE REVIEW

**Comment by gaetbout** [comment]
@LucasLvy

File: src/sorting/tests/bubble_sort_test.cairo [hunk 1]
```diff
@@ -21,22 +21,22 @@ fn bubblesort_test() {
...
+    let data = ArrayTrait::new();
 
-    let mut correct = ArrayTrait::new();
+    let correct = ArrayTrait::new();
 
-    let mut sorted = bubble_sort::bubble_sort_elements(data);
+    let sorted = bubble_sort::bubble_sort_elements(data);
 
-    assert(is_equal(sorted.span(), correct.span(), 0_u32) == true, 'invalid result');
+    assert(is_equal(sorted.span(), correct.span()) == true, 'invalid result');
```
Comment (LucasLvy) [inline src/sorting/tests/bubble_sort_test.cairo:39]:
```suggestion
    assert(is_equal(sorted.span(), correct.span()), 'invalid result');
```

File: src/sorting/tests/bubble_sort_test.cairo [hunk 2]
```diff
@@ -48,9 +48,9 @@ fn bubblesort_test_one_element() {
     let mut correct = ArrayTrait::new();
     correct.append(2_u32);
 
-    let mut sorted = bubble_sort::bubble_sort_elements(data);
+    let sorted = bubble_sort::bubble_sort_elements(data);
 
-    assert(is_equal(sorted.span(), correct.span(), 0_u32) == true, 'invalid result');
+    assert(is_equal(sorted.span(), correct.span()) == true, 'invalid result');
```
Comment (LucasLvy) [inline src/sorting/tests/bubble_sort_test.cairo:53]:
```suggestion
    assert(is_equal(sorted.span(), correct.span()), 'invalid result');
```

File: src/sorting/tests/bubble_sort_test.cairo [hunk 3]
```diff
@@ -21,22 +21,22 @@ fn bubblesort_test() {
     correct.append(4_u32);
     correct.append(5_u32);
 
-    let mut sorted = bubble_sort::bubble_sort_elements(data);
+    let sorted = bubble_sort::bubble_sort_elements(data);
 
-    assert(is_equal(sorted.span(), correct.span(), 0_u32) == true, 'invalid result');
+    assert(is_equal(sorted.span(), correct.span()) == true, 'invalid result');
```
Comment (LucasLvy) [inline src/sorting/tests/bubble_sort_test.cairo:26]:
```suggestion
    assert(is_equal(sorted.span(), correct.span()), 'invalid result');
```

**Review by LucasLvy — CHANGES_REQUESTED** [review]
you get the idea

File: src/math/signed_integers.cairo [hunk 4]
```diff
@@ -1119,7 +1119,7 @@ fn i33_le(lhs: i33, rhs: i33) -> bool {
 // # Returns
 // * `bool` - `true` if `lhs` is greater than or equal to `rhs`, `false` otherwise.
 fn i33_ge(lhs: i33, rhs: i33) -> bool {
-    if ((lhs == rhs) | i33_gt(lhs, rhs) == true) {
+    if (lhs == rhs) | i33_gt(lhs, rhs) {
```
Comment (LucasLvy) [inline src/math/signed_integers.cairo:1122]: replace function body by this ?
```suggestion
    (lhs == rhs) || i33_gt(lhs, rhs) 
```

**Review by LucasLvy — CHANGES_REQUESTED** [review]
you get the idea

File: src/math/tests/signed_integers_test.cairo [hunk 5]
```diff
@@ -190,14 +190,14 @@ fn i9_test_div_rem() {
     let b = i9 { inner: 5_u8, sign: true };
     let (q, r) = i9_div_rem(a, b);
     assert(q.inner == 2_u8 && r.inner == 3_u8, '-13 // -5 = 2 r -3');
-    assert(q.sign == false && r.sign == true, '-13 // -5 -> positive');
+    assert(q.sign == false && r.sign, '-13 // -5 -> positive');
```
Comment (LucasLvy) [inline src/math/tests/signed_integers_test.cairo:193]:
```suggestion
    assert(!q.sign && r.sign, '-13 // -5 -> positive');
```

**Review by LucasLvy — CHANGES_REQUESTED** [review]
you get the idea

File: src/math/tests/signed_integers_test.cairo (hunk 5 above)
Comment (gaetbout) [reply src/math/tests/signed_integers_test.cairo:193]: prefer seing == false in that case, the "!" can be not noticed

Comment (LucasLvy) [reply src/math/tests/signed_integers_test.cairo:193]: Hmm i don't think it's a problem and I don't think it's a good idea to encourage `== bool`

Comment (gaetbout) [reply src/math/tests/signed_integers_test.cairo]: Done everywhere then

**Review by LucasLvy — APPROVED** [review]
lgtm

**Pull request merged**